kolibri_instant_schools_plugin/assets/src/state/actions.js

```diff
@@ -113,7 +45,82 @@ function showAccount(store) {
   store.dispatch('CORE_SET_PAGE_LOADING', false);
   store.dispatch('CORE_SET_ERROR', null);
   store.dispatch('CORE_SET_TITLE', translator.$tr('userAccountPageTitle'));
-  resetAccountState(store);
+  const pageState = {
+    nameBusy: false,
+    nameSuccess: false,
+    nameError: false,
+    nameErrorMessage: '',
+    passwordError: false,
+    passwordBusy: false,
+    passwordSuccess: false,
+    passwordErrorMessage: '',
+    showPasswordModal: false,
+  };
+  store.dispatch('SET_PAGE_STATE', pageState);
+}
+
+function _errorMessageHandler(apiError) {
+  if (apiError.status.code === 400) {
+    return Object.values(apiError.entity)[0][0];
+  } else if (apiError.status.code === 403) {
+    return apiError.entity[0];
+  }
+  return '';
+}
+
+function changeName(store, newName) {
+  return FacilityUserProfileResource.getModel(store.state.core.session.user_id)
+    .save({ full_name: newName })
+    .then(
+      () => {
+        coreActions.getCurrentSession(store, true);
+        store.dispatch('SET_NAME_SUCCESS', true);
+        store.dispatch('SET_NAME_ERROR', false, '');
+        store.dispatch('SET_NAME_BUSY', false);
+      },
+      error => {
+        store.dispatch('SET_NAME_SUCCESS', false);
+        store.dispatch('SET_NAME_ERROR', true, _errorMessageHandler(error));
+        store.dispatch('SET_NAME_BUSY', false);
+      }
+    );
+}
+
+function changePassword(store, newPassword) {
```

Comment:
`changeName` and `changePassword` look almost the same. You could make this more generic and call it `changeFacilityUser` or something, and pass in the diff (e.g. `{ password: newPassword}`) into the `.save()` method.

Reply:
Hmm, but the store dispatches are different.

Reply:
ah, i didn't see that. for future reference, would it have been possible to manage the `change-password-modal` and `account-page`'s success/error/busy state within the component instead of in vuex? I see it done both ways in kolibri.

if so, we could utilize the generic `changeFacilityUser` and then handle the different paths within the component.

```
// in change-password-modal
// show busy
changeFacilityUser({ password: new_password })
.then(() => { ... show success })
.catch(() => { ... show failure })
```